src/Nox.Types/Types/Temperature/Temperature.cs

```diff
@@ -7,70 +9,41 @@ namespace Nox.Types;
 /// Represents a Nox <see cref="Temperature"/> type and value object.
 /// </summary>
 /// <remarks>Placeholder, needs to be implemented</remarks>
-public sealed class Temperature : ValueObject<(float TemperatureValue, TemperatureTypeUnit Unit), Temperature>
+public sealed class Temperature : Measurement<Temperature, TemperatureUnit>
 {
     /// <summary>
-    /// Absolute zero degrees represented in Fahrenheit.
+    /// Absolute zero degrees represented in Celsius.
     /// </summary>
-    private const float MinimumValueFahrenheit = -273.15f;
+    private const float MinimumValueCelsius = -273.15f;
     /// <summary>
-    /// Absolute zero degrees represented in Celsius.
+    /// Absolute zero degrees represented in Fahrenheit.
     /// </summary>
-    private const float MinimumValueCelsius = -459.67f;
-
-    public float TemperatureValue
-    {
-        get => Value.TemperatureValue;
-        private set => Value = (TemperatureValue: value, Value.Unit);
-    }
-    public TemperatureTypeUnit Unit
-    {
-        get => Value.Unit;
-        private set => Value = (Value.TemperatureValue, Unit: value);
-    }
-
-    public Temperature() { Value = (0, TemperatureTypeUnit.Celsius); }
+    private const float MinimumValueFahrenheit = -459.67f;
 
     /// <summary>
     /// Creates a new instance of <see cref="Temperature"/> object in <see cref="TemperatureTypeUnit.Celsius"/>.
     /// </summary>
     /// <param name="temperature">The value to create the <see cref="Temperature"/> with.</param>
     /// <returns></returns>
     /// <exception cref="TypeValidationException"></exception>
-    public static Temperature FromCelsius(float temperature) => From(temperature, TemperatureTypeUnit.Celsius);
+    public static Temperature FromCelsius(QuantityValue temperature) => From(temperature, TemperatureUnit.Celsius);
 
     /// <summary>
     /// Creates a new instance of <see cref="Temperature"/> object in <see cref="TemperatureTypeUnit.Fahrenheit"/>.
```

Comment:
You changed the type name and didn't update the comment.

Reply:
done